File: px-checkout/src/main/java/com/mercadopago/android/px/internal/features/express/slider/ConfirmButtonAdapter.java
package com.mercadopago.android.px.internal.features.express.slider;

import androidx.annotation.NonNull;
import android.view.View;
import com.mercadopago.android.px.internal.features.pay_button.PayButtonFragment;
import com.mercadopago.android.px.internal.viewmodel.ConfirmButtonViewModel;
import com.mercadopago.android.px.internal.viewmodel.SplitSelectionState;
import com.mercadopago.android.px.model.internal.Application;
import java.util.List;

public class ConfirmButtonAdapter extends HubableAdapter<List<ConfirmButtonViewModel>, View> {

    private PayButtonFragment payButton;

    public ConfirmButtonAdapter(@NonNull final PayButtonFragment fragment) {
        super(null);
        payButton = fragment;
    }

    @Override
    public void updateData(final int currentIndex, final int payerCostSelected,
        @NonNull final SplitSelectionState splitSelectionState,
        final @NonNull Application application) {
        if (data.get(currentIndex).isDisabled()) {
            payButton.disable();
        } else {
            payButton.enable();
        }
    }

    @Override
    public List<ConfirmButtonViewModel> getNewModels(final HubAdapter.Model model) {
        return model.confirmButtonViewModels;
    }
}

File: px-checkout/src/main/java/com/mercadopago/android/px/internal/features/express/slider/ConfirmButtonAdapter.kt
package com.mercadopago.android.px.internal.features.express.slider

import android.view.View
import com.mercadopago.android.px.internal.features.pay_button.PayButtonFragment
import com.mercadopago.android.px.internal.viewmodel.ConfirmButtonViewModel
import com.mercadopago.android.px.internal.viewmodel.SplitSelectionState
import com.mercadopago.android.px.model.internal.Application

internal class ConfirmButtonAdapter(private val payButton: PayButtonFragment)
    : HubableAdapter<List<ConfirmButtonViewModel.ByApplication>, View>(null) {

    override fun updateData(currentIndex: Int, payerCostSelected: Int,
        splitSelectionState: SplitSelectionState, application: Application) {
        data?.let {
            if (it[currentIndex][application].isDisabled) payButton.disable() else payButton.enable()
        }
    }

    override fun getNewModels(model: HubAdapter.Model): List<ConfirmButtonViewModel.ByApplication> {
        return model.confirmButtonViewModels
    }

}